ros2bag/ros2bag/verb/record.py

```diff
@@ -55,7 +55,7 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
         parser.add_argument(
             '-e', '--regex', default='',
             help='Record only topics containing provided regular expression. '
-            'Overrides --all, applies on top of topics list.')
+            'Overridden by --all, applies on top of topics list.')
```

Comment:
@Barry-Xu-2018 IMO `Overridden by --all` sounds a bit confusing. 
I think it would be better to remove it from the help section at all. Just keep `applies on top of topics list.`
And add a check in the main function with a warning message like
```
if args.all and args.regex:
  print_warn('--all will override --regex')
```

Reply:
Yes. This way is clearer than writing it in the parameter description.   
I will change it.

Reply:
Fixed in https://github.com/ros2/rosbag2/pull/1644/commits/c84824fc92e0a4b43b0d0a0c4f0c443c9cbf3279 

Next, I will create a new PR for Iron based on this PR and close this PR. After PR for Iron is merged, backport it to Humble.